### PR TITLE
feature: custom databases support

### PR DESCRIPTION
**Pull request by dudzicp**

### CODE REVIEW

**Comment by BartoszJanuszNA** [comment]
Commit name should not be `feature` but `feat`

**Review by DawidChomiczNA — APPROVED** [review]
LGTM, although it should have tests added soon after publication is released

File: riot_na/airr/airr_builder.py [hunk 1]
```diff
@@ -263,54 +343,75 @@ def _extract_region(self, start: int, end: int):
...
         )
 
-        if self.rearrangement.sequence_alignment_aa:
-            alignment_offset = self.v_gene_alignment.q_start + self.rearrangement.v_frame
-            v_start_aa = 0
-            v_end_aa = int((self.v_gene_alignment.q_end - alignment_offset) / 3)
+        germline_segment = v_germline_segment + d_germline_segment + j_germline_segment
 
-            self.rearrangement.v_sequence_alignment_aa = self.rearrangement.sequence_alignment_aa[v_start_aa:v_end_aa]
+        v_aln_str = unfold_cigar(self.v_gene_alignment.cigar)
```
Comment (BartoszJanuszNA) [inline riot_na/airr/airr_builder.py:383]: Unfolding over and over, assign to self and reuse.

File: riot_na/airr/airr_builder.py [hunk 2]
```diff
@@ -263,54 +343,75 @@ def _extract_region(self, start: int, end: int):
...
+            )
+            dj_junction_alignment_str = dj_junction_len * "I"
 
-                self.rearrangement.j_sequence_alignment_aa = self.rearrangement.sequence_alignment_aa[
-                    j_start_aa:j_end_aa
-                ]
+            vj_junction_alignment_str = vd_junction_alignment_str + d_aln_str + dj_junction_alignment_str
 
+        full_alignment_str = v_aln_str + vj_junction_alignment_str + j_aln_str
+
```
Comment (BartoszJanuszNA) [inline riot_na/airr/airr_builder.py:407]: Extract to function

File: riot_na/airr/airr_builder_aa.py [hunk 3]
```diff
@@ -158,18 +174,38 @@ def with_scheme_alignment_exc(self, exc: str):
...
+
+        v_aln_str = unfold_cigar(self.v_gene_alignment_aa.cigar)
+        j_aln_str = unfold_cigar(self.j_gene_alignment_aa.cigar) if self.j_gene_alignment_aa else ""
+
+        vj_junction_len = (
+            self.j_gene_alignment_aa.q_start - self.v_gene_alignment_aa.q_end if self.j_gene_alignment_aa else 0
+        )
+        vj_junction_alignment_str = vj_junction_len * "I"
+
+        full_alignment_str = v_aln_str + vj_junction_alignment_str + j_aln_str
```
Comment (BartoszJanuszNA) [inline riot_na/airr/airr_builder_aa.py:202]: Duplication

File: riot_na/alignment/nt_gene_alignments.py [hunk 4]
```diff
@@ -104,9 +104,9 @@ def create_vdjc_aligner_nt(allowed_species: Optional[list[Organism]] = None, db_
     c_aligners = {}
 
     for locus in Locus:
-        c_aligner = create_aligner(
-            allowed_species=Organism.HOMO_SAPIENS, germline_gene=GermlineGene.C, locus=locus, db_dir=db_dir
-        )
+        c_organism = Organism.HOMO_SAPIENS if organism != Organism.CUSTOM else Organism.CUSTOM
```
Comment (BartoszJanuszNA) [inline riot_na/alignment/nt_gene_alignments.py:107]: What if custom db does not have C genes?

File: riot_na/api/riot_numbering.py [hunk 5]
```diff
@@ -234,7 +227,7 @@ def run_on_sequence(
 
         if alignments.v:
             airr_builder.with_v_gene_alignment_aa(alignments.v)
-            airr_builder.with_sequence_alignment_aa(aligned_sequence)
+            # airr_builder.with_sequence_alignment_aa(aligned_sequence)
```
Comment (BartoszJanuszNA) [inline riot_na/api/riot_numbering.py:230]: ? remove if unnecesary

**Review by BartoszJanuszNA — COMMENTED** [review]
In notebook use markdown and headers instead of plain comments -> more readable

**Pull request merged**